aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -199,8 +205,34 @@ private static String getDataJson(Subscription subscription) {
         }
     }
 
-    private static String getApiUrl(AWSConfiguration awsConfiguration) throws JSONException {
-        return awsConfiguration.optJsonObject("AppSync").getString("ApiUrl");
+    private String getRegion() throws JSONException {
+         return mRegionStr != null
+                 ? mRegionStr
+                 : mAwsConfiguration
+                     .optJsonObject("CredentialsProvider")
+                     .getJSONObject("CognitoIdentity")
+                     .getJSONObject(mAwsConfiguration.getConfiguration())
+                     .getString("Region");
+    }
+
+    private String getIdentityPoolId() throws JSONException {
+         return mAwsConfiguration
+                 .optJsonObject("CredentialsProvider")
+                 .getJSONObject("CognitoIdentity")
+                 .getJSONObject(mAwsConfiguration.getConfiguration())
+                 .getString("PoolId");
+    }
+
+    private String getApiUrl() throws JSONException {
+        return mServerUrl != null
+                ? mServerUrl
+                : mAwsConfiguration.optJsonObject("AppSync").getString("ApiUrl");
```

Comment:
This seems to be redundant after [these changes in the AppSyncClient](https://github.com/awslabs/aws-mobile-appsync-sdk-android/pull/276/files#diff-980fbeba2fd8c9fcedb39b427e501a6cR554)?

Reply:
Nice catch, I think this is no longer needed. I'll update my PR 👍